### PR TITLE
Add Semgrep CI YAML file

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [ ] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Review by holdenmcg — APPROVED** [review]

File: .github/workflows/semgrep.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Review CI configuration
+Review our semgrep.yml config file.
```
Comment (pabloest) [inline .github/workflows/semgrep.yml:2]: Are these comments? If so, shouldn’t they be commented lines?

File: .github/workflows/semgrep.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    paths:
+      - .github/workflows/semgrep.yml
+  schedule:
+    - cron: '0 0 * * 0'
+jobs:
+  semgrep:
+    name: Scan
+    runs-on: ubuntu-20.04
+    env:
+      SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
```
Comment (pabloest) [inline .github/workflows/semgrep.yml:20]: If this connects to Semgrep App, won’t the whole repo be connected and follow the Semgrep App rule board that is used by all of our org?

Comment (adamkvitek) [reply .github/workflows/semgrep.yml:20]: Yes, there are workarounds though how to apply only specific rules for us.

File: .github/workflows/semgrep.yml (hunk 1 above)
Comment (adamkvitek) [reply .github/workflows/semgrep.yml:2]: True. Editing.

**Comment by adamkvitek** [comment]
@pabloest should we rather create another org in App only for our docs? I feel that our generic mode rules checking for language and Markdown/HTML stylistics may create a lot of noise in other repositories bound to the current org.

**Comment by adamkvitek** [comment]
Agreed with @pabloest - we will rather create our local YAML file with specific rules - so not all r2c organization rules are applied to us (and vice versa).

**Comment by adamkvitek** [comment]
See: https://github.com/returntocorp/semgrep-docs/pull/618#issuecomment-1234476100

**Pull request closed without merging**